File: FundamentosDeLaProgramacionLuisJoyanes4taEdicion/chapter03/JAVA/LogicalExpressions.java
/*Libro:Fundamentos de Programación, Algoritmos, estructura de datos y objetos 4ta edición. 
    Capitulo 03.-Estructura general de un programa.
    Problema 12.-resume una serie de aplicaciones de expresiones lógicas.
					
	Expresión lógica                    Resultado                   Observaciones
    (1 > 0) y (3 = 3)                   verdadero
    no PRUEBA                           verdadero                   ∙PRUEBA es un valor lógico falso.
    (0 < 5) o (0 > 5)                   verdadero
    (5 <= 7) y (2 > 4)                  falso
    no (5 <> 5)                         verdadero
    (numero = 1) o (7 >= 4)             verdadero                   ∙numero es una variable entera de valor 5.
*/

public class LogicalExpressions{
    public static void main(String[] args){
        boolean PRUEBA=false;
        int number=5;

        System.out.print("   Expresion logica\t\tResultado\t\tObservaciones");

        if((1 > 0) && (3 == 3)){
            System.out.print("\n\n   (1 > 0) && (3 == 3)\t\tVerdadero\n");
        }else{
            System.out.print("\n   (1 > 0) && (3 == 3)\t\tFalso\n");
        }

        if((!PRUEBA)){
            System.out.print("\n   no PRUEBA\t\t\tVerdadero\t\tPRUEBA es un valor logico falso\n");
        }else{
            System.out.print("\n   no PRUEBA\t\tFalso\n");
        }

        if((0 < 5) || (0 > 5)){
            System.out.print("\n   (0 < 5) o (0 > 5)\t\tVerdadero\n");
        }else{
            System.out.print("\n   (0 < 5) o (0 > 5)\t\tFalso\n");
        }

        if((5 <= 7) && (2 > 4)){
            System.out.print("\n   (5 <= 7) && (2 > 4)\t\tVerdadero\n");
        }else{
            System.out.print("\n   (5 <= 7) && (2 > 4)\t\tFalso\n");
        }

        if(!(5!=5)){
            System.out.print("\n   !(5<>5)\t\t\tVerdadero\n");
        }else{
            System.out.print("\n   !(5<>5)\t\t\tFalso\n");
        }

        if((number==5) || (7>=4)){
            System.out.print("\n   (numero==5) || (7>=4)\tVerdadero\t\tnumero es una variable entera de valor 5");
        }else{
            System.out.print("\n   (numero==5) || (7>=4)\t\tFalso");
        }
    
        System.out.print("\n\n");
    }   
}

File: FundamentosDeLaProgramacionLuisJoyanes4taEdicion/chapter03/JAVASCRIPT/js/logicalExpressions.js
function calculate(){
    
    if(document.getElementById("idBtnCalculate").value=="Calcular"){                 
        
        let tableBody=document.createElement("tbody");
        tableBody.setAttribute("id", "idTableBody");     

        var PRUEBA=false;
        var number=5;

        if((1 > 0) && (3 == 3)){
            tableBody.innerHTML+="<tr><td> (1 > 0) && (3 == 3) </td><td> Verdadero </td><td> </td></tr>";
        }else{
            tableBody.innerHTML+="<tr><td> (1 > 0) && (3 == 3) </td><td> Falso </td><td> </td></tr>";
        }

        if((!PRUEBA)){
            tableBody.innerHTML+="<tr><td> no PRUEBA </td><td> Verdadero </td><td>es un valor logico falso </td></tr>";
            
        }else{
            tableBody.innerHTML+="<tr><td> no PRUEBA </td><td> Falso </td><td>es un valor logico falso </td></tr>";
        }
        
        if((0 < 5) || (0 > 5)){
            tableBody.innerHTML+="<tr><td> (0 < 5) o (0 > 5) </td><td> Verdadero </td><td> </td></tr>";
        }else{
            tableBody.innerHTML+="<tr><td> (0 < 5) o (0 > 5) </td><td> Falso </td><td> </td></tr>";
        }

        if((5 <= 7) && (2 > 4)){
            tableBody.innerHTML+="<tr><td>  (5 <= 7) && (2 > 4) </td><td> Verdadero </td><td> </td></tr>";
        }else{
            tableBody.innerHTML+="<tr><td>  (5 <= 7) && (2 > 4) </td><td> Falso </td><td> </td></tr>";
        }

        if(!(5!=5)){
            tableBody.innerHTML+="<tr><td>  !(5<>5) </td><td> Verdadero </td><td> </td></tr>";
        }else{
            tableBody.innerHTML+="<tr><td>  !(5<>5) </td><td> Falso </td><td> </td></tr>";
        }

        if((number==5) || (7>=4)){
            tableBody.innerHTML+="<tr><td>  (numero==5) || (7>=4) </td><td> Verdadero </td><td> Numero es una variable entera de valor 5 </td></tr>";
        }else{
            tableBody.innerHTML+="<tr><td>  (numero==5) || (7>=4) </td><td> Falso </td><td> Numero es una variable entera de valor 5 </td></tr>";
        }
    
        document.getElementById("idTable").appendChild(tableBody);
    
        document.getElementById("idBtnCalculate").value="Limpiar";
    }else{
        document.getElementById("idTableBody").remove();
        
        document.getElementById("idBtnCalculate").value="Calcular";
    }
    
        
        
}